### PR TITLE
Script to found missing or unused i18n placeholders

### LINKED ISSUE #252: options.common.down and options.common.up never used
I've search for options.common.down and options.common.up 
Nowhere I found this calls 

but in I18n  there  are  .up and .down in common section

### PR DESCRIPTION
**Pull request by Moosh-be**
see https://github.com/bluecaret/carettab/issues/252#issuecomment-963359341

Script to found missing or unused i18n placeholders
- V1. provide a list of placehoder found nowhere in code
- V2.  (later)  provide a list of placehoder found in code but missing in i18n files
- V3 use full path of place holder (not only final part)

Unfortunately write in bash not in js.

I would use it to solve #252   but I suppose that tool in bash is not welcome.

Option 1 : I keep it in a gist
Option 2 : I found help to rewrite it in js.

### CODE REVIEW

File: checkI18nVars.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        console.log(err)
+        return
+    }
+
+    // purée ca marche mais un bon codeur js va avoir une crise cardiaque en voyant ca ci dessous.
+
+    let arr= [];
+    for (let key of Object.keys(i18nFile)) {
+        if (isObject(i18nFile[key])) 
+        for (let key2 of Object.keys(i18nFile[key])) {
```
Comment (Moosh-be) [inline checkI18nVars.js:37]: il y a moyen de faire une fonction récursive.
mais je n'aime pas la récursivité

File: checkI18nVars.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        try {
+            const object = JSON.parse(fileData)
+            return cb && cb(null, object)
+        } catch (err) {
+            return cb && cb(err)
+        }
+    })
+} 
+
+function pushAndLog(key, val, arr) {
```
Comment (Moosh-be) [inline checkI18nVars.js:21]: j'empile les clés dans arr, ca me dérange de passer arr comme ca mais ca marche

**Comment by bluecaret** [comment]
Thank you for this! I hope to take a look at this soon! Bash or JS works for me for this.

**Comment by Moosh-be** [comment]
The Js version was too ambitious for me.

I think  just use my bash to detect changes to propose, and publish the bash in a gist.

**Pull request closed without merging**